File: app/src/main/java/com/goranch/publicapis/data/DbUtil.java
package com.goranch.publicapis.data;

public class DbUtil {
    public static String strSeparator = "__,__";

    public static String convertArrayToString(String[] array) {
        StringBuilder str = new StringBuilder();
        for (int i = 0; i < array.length; i++) {
            str.append(array[i]);
            // Do not append comma at the end of last element
            if (i < array.length - 1) {
                str.append(strSeparator);
            }
        }
        return str.toString();
    }

    public static String[] convertStringToArray(String str) {
        return str.split(strSeparator);
    }
}

File: app/src/main/java/com/goranch/publicapis/data/DbUtil.kt
package com.goranch.publicapis.data

object DbUtil {
    var strSeparator = "__,__"

    fun convertArrayToString(array: Array<String>): String {
        val str = StringBuilder()
        for (i in array.indices) {
            str.append(array[i])
            // Do not append comma at the end of last element
            if (i < array.size - 1) {
                str.append(strSeparator)
            }
        }
        return str.toString()
    }

    fun convertStringToArray(str: String): Array<String> {
        return str.split(strSeparator.toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
    }
}
